From akpm@linux-foundation.org  Tue Sep 23 15:49:46 2008
From: SangSu Park<sangsu@gmail.com>
Date: Mon, 22 Sep 2008 15:41:15 -0700
Subject: USB: g_printer: fix handling zero-length packet
To: mm-commits@vger.kernel.org
Cc: sangsu@gmail.com, david-b@pacbell.net, greg@kroah.com, stern@rowland.harvard.edu
Message-ID: <200809222241.m8MMfFxF001842@imap1.linux-foundation.org>

From: SangSu Park<sangsu@gmail.com>

g_printer doesn't have to check whether the data size is a multiple of
MaxPacketSize, because device controller driver already make that check.

Signed-off-by: SangSu Park<sangsu@gmail.com>
Cc: Alan Stern <stern@rowland.harvard.edu>
Cc: David Brownell <david-b@pacbell.net>
Cc: Greg KH <greg@kroah.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/printer.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/usb/gadget/printer.c
+++ b/drivers/usb/gadget/printer.c
@@ -1278,8 +1278,7 @@ unknown:
 	/* respond with data transfer before status phase? */
 	if (value >= 0) {
 		req->length = value;
-		req->zero = value < wLength
-				&& (value % gadget->ep0->maxpacket) == 0;
+		req->zero = value < wLength;
 		value = usb_ep_queue(gadget->ep0, req, GFP_ATOMIC);
 		if (value < 0) {
 			DBG(dev, "ep_queue --> %d\n", value);
